src/java/org/apache/cassandra/io/compress/CompressedSequentialWriter.java

```diff
@@ -332,6 +332,41 @@ private void seekToChunkStart()
         }
     }
 
+    // Page management using chunk boundaries
+
+    @Override
+    public int maxBytesInPage()
+    {
+        return buffer.capacity();
+    }
+
+    @Override
+    public void padToPageBoundary() throws IOException
+    {
+        if (buffer.position() == 0)
+            return;
+
+        int padLength = buffer.remaining();
```

Comment:
```suggestion
        int padLength = bytesLeftInPage();
```
I wonder if this would read better.